lib/datadog/core/diagnostics/environment_logger.rb

```diff
@@ -221,9 +221,9 @@ def health_metrics_enabled
           !!Datadog.configuration.diagnostics.health_metrics.enabled
         end
 
-        # TODO: Populate when profiling is implemented
-        # def profiling_enabled
-        # end
+        def profiling_enabled
+          !!Datadog.configuration.profiling.enabled
```

Comment:
We don't need `!!`, since the option is already coerced with `env_bool`.

```
         option :enabled do |o|
            o.default { env_to_bool(Profiling::Ext::ENV_ENABLED, false) }
            o.lazy
          end
```

Reply:
Can someone do `c.profiling.enabled = nil`?

Reply:
Oh yeah, technically it can be set to any value. but I think our [document](https://docs.datadoghq.com/profiler/enabling/ruby/?tab=incode) make it clear this is a boolean and [Yard doc](https://github.com/DataDog/dd-trace-rb/blob/master/lib/datadog/core/configuration/settings.rb#L193) is being consistent with that.

Reply:
Good point. It turns out you can use whatever you want:

```
$ DD_TRACE_DEBUG=true bundle exec pry
[1] pry(main)> require 'ddtrace'
=> true
[2] pry(main)> Datadog.configure { |c| c.profiling.enabled = 'i like pie' };
D, [2023-01-12T13:52:08.108222 #628982] DEBUG -- ddtrace: [ddtrace] (dd-trace-rb/lib/datadog/core/workers/async.rb:132:in `start_worker') Starting thread for: #<Datadog::Core::Telemetry::Heartbeat:0x000062ba9b97a3e8>
D, [2023-01-12T13:52:08.108383 #628982] DEBUG -- ddtrace: [ddtrace] (dd-trace-rb/lib/datadog/core/configuration/components.rb:405:in `startup!') Profiling started
```

So I think the `!!` is probably best to keep for now, although we should probably think about making the configuration system itself less permissive.